param/parameterized.py

```diff
@@ -719,34 +719,82 @@ class Foo(Bar):
 
-    def __init__(self,default=None,doc=None,label=None,precedence=None,  # pylint: disable-msg=R0913
-                 instantiate=False,constant=False,readonly=False,
+    def __init__(self,default=None, doc=None, label=None, precedence=None,  # pylint: disable-msg=R0913
+                 instantiate=False, constant=False, readonly=False,
                  pickle_default_value=True, allow_None=False,
                  per_instance=True):
-        """
-        Initialize a new Parameter object: store the supplied attributes.
-
-        default: the owning class's value for the attribute
-        represented by this Parameter.
-
-        precedence is a value, usually in the range 0.0 to 1.0, that
-        allows the order of Parameters in a class to be defined (for
-        e.g. in GUI menus). A negative precedence indicates a
-        parameter that should be hidden in e.g. GUI menus.
 
-        default, doc, and precedence default to None. This is to allow
+        """Initialize a new Parameter object and store the supplied attributes:
+
+        default: the owning class's value for the attribute represented
+        by this Parameter, which can be overridden in an instance.
+
+        doc: docstring explaining what this parameter represents.
+
+        label: optional text label to be used when this Parameter is
+        shown in a listing. If no label is supplied, the attribute name
+        for this parameter in the owning Parameterized object is used.
+
+        precedence: a numeric value, usually in the range 0.0 to 1.0,
+        which allows the order of Parameters in a class to be defined in
+        a listing or e.g. in GUI menus. A negative precedence indicates
+        a parameter that should be hidden in such listings.
+
+        instantiate: controls whether the value of this Parameter will
+        be deepcopied when a Parameterized object is instantiated (if
+        True), or if the single default value will be shared by all
+        Parameterized instances (if False). For an immutable Parameter
+        value, it is best to leave instantiate at the default of
+        False, so that a user can choose to change the value at the
+        Parameterized instance level (affecting only that instance) or
+        at the Parameterized class or superclass level (affecting all
+        instances of that class or superclass). For a mutable
+        Parameter value, the default of False is also appropriate if
+        you want all instances to share the same value state, e.g. if
+        they are each simply referring to a single global object like
+        a singleton. If each Parameterized should have its own
+        independently mutable value, instantiate should be set to
+        True, but note that there is then no simple way to change the
+        value of this Parameter at the class or superclass level,
+        because each instance will now have an independently
+        instantiated value.
```

Comment:
Changing the default will still affect instances created after making that change right? Not sure if that's worth clarifying though.

Reply:
True. Feel free to propose a change to say that!

Reply:
Each *existing* instance?

Reply:
Yeah, that works.